app.js

```diff
@@ -0,0 +1,21 @@
+const express = require('express');
+const { getAllTopics } = require('./controllers/topics.controllers');
+const app = express();
+app.use(express.json())
+
+app.get('/api/topics', getAllTopics);
+
+
+
+
+
+
+
+
+
+app.use((err, req, res, next) => {
+    console.log(err)
+    res.status(404).send({msg:'Bad request'})
```

Comment:
in what situation would the middleware pick up that it's a 404 here as opposed to any other error? have a think about how you would change this when you want to account for other types of errors in future